tools/provision.sh

```diff
@@ -143,17 +143,17 @@ else # AUTH_METHOD_NONE
 
     # RHEL nightly repos need to be overridden
     case "${ID}-${VERSION_ID}" in
-        "rhel-8.6")
-            # Override old rhel-8.json and rhel-8-beta.json because RHEL 8.6 test needs nightly repos
-            sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-86.json "$REPODIR/rhel-8.json"
+        "rhel-8.7")
+            # Override old rhel-8.json and rhel-8-beta.json because RHEL 8.7 test needs nightly repos
+            sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-87.json "$REPODIR/rhel-87.json"
             # If multiple tests are run and call provision.sh the symlink will need to be overridden with -f
-            sudo ln -sf /etc/osbuild-composer/repositories/rhel-8.json "$REPODIR/rhel-8-beta.json"
+            sudo ln -sf /etc/osbuild-composer/repositories/rhel-87.json "$REPODIR/rhel-87-beta.json"
             ;;
-        "rhel-9.0")
-            # Override old rhel-90.json and rhel-90-beta.json because RHEL 9.0 test needs nightly repos
-            sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-90.json "$REPODIR/rhel-90.json"
+        "rhel-9.1")
+            # Override old rhel-91.json and rhel-91-beta.json because RHEL 9.1 test needs nightly repos
+            sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-91.json "$REPODIR/rhel-91.json"
             # If multiple tests are run and call provision.sh the symlink will need to be overridden with -f
-            sudo ln -sf /etc/osbuild-composer/repositories/rhel-90.json "$REPODIR/rhel-90-beta.json"
+            sudo ln -sf /etc/osbuild-composer/repositories/rhel-91.json "$REPODIR/rhel-91-beta.json"
             ;;
         *) ;;
     esac
```

Comment:
So I checked the script in more detail.

This whole block can be IMO removed, because it is useless. The same files are already copied over to the `$REPODIR` above this block, so copying them again does not add any value.

In addition, `-beta.json` repos are nor used in any case, even for Beta.

Reply:
@thozza what about https://github.com/osbuild/osbuild-composer/blob/main/test/cases/regression-composer-works-behind-satellite-fallback.sh#L68 ? 

If CI jobs don't fail I will remove those references to -beta.json as well.

Reply:
@atodorov that should be the same case. So yes, IMO it can be removed as well. The current composer implementation will never pick the `-beta.json` repo definitions, even if running on Beta release.

In short:

1. When composer starts, it constructs the list of supported distros and determines on which distro it is running:
https://github.com/osbuild/osbuild-composer/blob/13e287733a07873f26b311fc9a86692b6bad7958/internal/distroregistry/distroregistry.go#L69

2. It then calls `New*HostDistro()` on the appropriate distro implementation, while passing in the information if the distro is Beta: https://github.com/osbuild/osbuild-composer/blob/13e287733a07873f26b311fc9a86692b6bad7958/internal/distroregistry/distroregistry.go#L75

3. However, all `New*HostDistro()` implementations are ignoring the passed values and always return the "base" distro: https://github.com/osbuild/osbuild-composer/blob/13e287733a07873f26b311fc9a86692b6bad7958/internal/distro/rhel8/distro.go#L741-L743

4. Weldr API then uses `RepoRegistry` to load all repo definitions and uses the filename as the distro name:https://github.com/osbuild/osbuild-composer/blob/13e287733a07873f26b311fc9a86692b6bad7958/internal/rpmmd/repository.go#L293

5. And the `RepoRegistry` implementation will always look for the repos using the value returned by distro implementation's `Name()` method, which will never contain the `-beta` suffix: https://github.com/osbuild/osbuild-composer/blob/13e287733a07873f26b311fc9a86692b6bad7958/internal/reporegistry/reporegistry.go#L44